packages/grafana-data/src/types/logs.ts

```diff
@@ -260,3 +260,37 @@ export const hasLogsContextUiSupport = (datasource: unknown): datasource is Data
 
   return withLogsSupport.getLogRowContextUi !== undefined;
 };
+
+export interface AnalyzeQueryOptions {
+  check: string;
+  attributes: KeyValue<string>;
+}
+
+/**
+ * Data sources that support query manipulation through `modifyQuery` and `analyzeQuery` in Explore.
+ * Allows for interactions such as changing the query from logs details, or displaying filter status.
+ * @internal
+ */
+export interface DataSourceWithQueryManipulationSupport<TQuery extends DataQuery, TAnalyzeQueryResult = boolean> {
+  /**
+   * Used in explore
+   */
+  modifyQuery(query: TQuery, action: QueryFixAction): TQuery;
+
+  /**
+   * Used in explore for Log details
+   *
+   * @alpha
+   */
+  analyzeQuery?(query: TQuery, options: AnalyzeQueryOptions): TAnalyzeQueryResult;
```

Comment:
Just a question, but have you considered `analyzeQuery` being its own data source extension - `DataSourceWithQueryAnalyze...`. Or in other words, is there a place where this would be useful that is not tied to `modifyQuery`?

Reply:
I think both features are closely related and I can't foresee, at least in the logs space, the need to have them independently. YAGNI?